### DISCUSSION TITLE
How to remove a specific Tailwind utility class or css declaration from a daisyUI class?

### DISCUSSION THREAD

**Discussion by HappyCodingCoding** [saadeghi/daisyui#2728] [Q&A] [answered]
For example, the daisyUI class '.btn' defines a CSS declaration 'height: 3rem;', as seen in the picture below:
![image](https://github.com/saadeghi/daisyui/assets/149742267/6557a148-c63f-47dd-bd62-282dbbd4c2ce)

How can I remove the key-value pair ''height: 3rem' from the default daisyUI class '.btn'?
**Note,** I mean to remove it not override the value of the CSS declaration.

Thank you!

**Comment by saadeghi** [comment, marked as answer]
Why? What is the issue with overriding the style?

**Reply by HappyCodingCoding** [reply]
Just want the element's height not to be set so that it can auto fit the content within it .

**Reply by saadeghi** [reply]
If you want height to be auto, simply add `h-auto` and if you want to remove minimum height as well, also add `min-h-0`  
Example: https://play.tailwindcss.com/OirlfJcRwk

**Reply by dominikkrulak** [reply]
Of course this one is a solution, but that doesn't remove `height` css property from `.btn` class if that what @HappyCodingCoding was asking.

**Reply by saadeghi** [reply]
You're correct. It doesn't **_remove_** the default `height`  
That's why I asked @HappyCodingCoding what is it that they want to do, and they said they want the `height` to be auto. so if they want the height to be auto, `h-auto` does that. no need to fully remove the default value which would be harder to do (can be done with forking the package as you mentioned, or can be done with additional PostCSS plugins)

**Comment by dominikkrulak** [comment]
1. I think you'd have to fork the repository, remove the utility class and build it. 
2. If I remember it right you could extend it with your own button component defined in `tailwind.config.js` file.
3. Edit `tailwind.config.js` and set `utils` key in `daisyui` object literal to `false` and build the button from scratch in you  _css_ file like so `.btn {@apply p-4;}`. [The link to config documentation](https://daisyui.com/docs/config/).

I haven't tried any of this but at least 1. and 3. option could work.

**Reply by HappyCodingCoding** [reply]
Thanks for your reply.

**Comment by HappyCodingCoding** [comment]
On Mon, Jan 15, 2024 at 22:06 Pouya Saadeghi ***@***.***>
wrote:

> You're correct. It doesn't *remove* the default height
> That's why I asked @HappyCodingCoding
> <https://github.com/HappyCodingCoding> what is it that they want to do,
> and they said they want the height to be auto. so if they want the height
> to be auto, h-auto does that. no need to fully remove the default value
> which would be harder to do (can be done with forking the package as you
> mentioned, or can be done with additional PostCSS plugins)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/saadeghi/daisyui/discussions/2728#discussioncomment-8133490>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BDWOFO5VCYXZA22WKTZFZZDYOUZWXAVCNFSM6AAAAABBMK3QRKVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DCMZTGQ4TA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
Thank you very much! I’m the problem itself because I’m not very familiar
with CSS, let alone daisyUI.
